File: frontend_taller/lib/models/assistance_request.dart
class AssistanceRequest {
  final int id;
  final String clientName;
  final String vehiclePlate;
  final String location;
  final String description;
  final int providerId;
  final String status;
  final String? assignedTechnician;
  final Map<String, dynamic>? provider;

  AssistanceRequest({
    required this.id,
    required this.clientName,
    required this.vehiclePlate,
    required this.location,
    required this.description,
    required this.providerId,
    required this.status,
    this.assignedTechnician,
    this.provider,
  });

  factory AssistanceRequest.fromJson(Map<String, dynamic> json) {
    return AssistanceRequest(
      id: json['id'],
      clientName: json['client_name'],
      vehiclePlate: json['vehicle_plate'],
      location: json['location'],
      description: json['description'],
      providerId: json['provider_id'],
      status: json['status'],
      assignedTechnician: json['assigned_technician'],
      provider: json['provider'],
    );
  }

  String get providerName {
    if (provider == null) return 'No disponible';
    return provider!['name'] ?? 'No disponible';
  }
}

File: backend_taller/app/Models/AssistanceRequest.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class AssistanceRequest extends Model
{
    protected $fillable = [
        'client_name',
        'vehicle_plate',
        'location',
        'description',
        'provider_id',
        'status',
        'assigned_technician',
        'processed_at',
    ];

    protected $casts = [
        'processed_at' => 'datetime',
    ];

    public function provider(): BelongsTo
    {
        return $this->belongsTo(Provider::class);
    }
}